rts/Game/UI/SelectionKeyHandler.cpp

```diff
@@ -432,6 +432,27 @@ void CSelectionKeyHandler::DoSelection(std::string selectString)
 			return;
 		} break;
 
+		case hashString("SelectOneNearestMouse"): {
+			if(selection.empty())
+				return;
+
+			const float gndDist = CGround::LineGroundCol(camera->GetPos(), camera->GetPos() + mouse->dir * camera->GetFarPlaneDist(), false);
+			float3 mousePosition = camera->GetPos() + mouse->dir * gndDist;
+
+			CUnit* closest = nullptr;
+			float closestDistance = 0;
+			for(auto ui = selection.begin(); ui != selection.end(); ui++) {
+				CUnit* unit = *ui;
+				float distance = mousePosition.SqDistance(unit->pos);
+				if (!closest || distance < closestDistance) {
+					closestDistance = distance;
+				closest = unit;
+				}
+			}
+			if(closest)
+				selectedUnitsHandler.AddUnit(closest);
```

Comment:
Can't check it easily at the moment, but the semantic of the call suggests that `closest` unit will be added to the existing set of selected units. Is it what you want this call to do (vs dropping the current selection and select the nearest unit)?

Reply:
Yes, this should be decided by using `ClearSelection` or not in the full selector.